explorer-ui/src/components/Mapping/MapContainer.jsx

```diff
@@ -8,155 +8,265 @@ import {
+  const [activeMarker, setActiveMarker] = useState();
+  const [selectedPlace, setSelectedPlace] = useState();
+  const [activeProfile, setActiveProfile] = useState(false);
+  const [currProfile, setCurrProfile] = useState(null);
+  const origin = {lat: 37.550201, lng: -121.980827}
+  const originString= "Facebook, Fremont Street, San Francisco, CA, USA"
+  const onMarkerClick = (user, marker) => {
+    setActiveProfile(true);
+    calculateRouteBetweenUsers(user.user.address);
+    setSelectedPlace(user.user.address);
+    setActiveMarker(marker);
+    setshowingInfoWindow(true);
+    setCurrProfile(user);
+    console.log("curr profile is changing state: ", currProfile)
 
-  const [map, setMap] = useState(/** @type google.maps.Map */ (null))
-  const [directionsResponse, setDirectionsResponse] = useState(null)
-  const [distance, setDistance] = useState('')
-  const [duration, setDuration] = useState('')
+  };
+  // const onMouseOver = (profile, marker) => {
+  //   console.log("user address: ", profile.address)
+  //   setSelectedPlace(profile.user.address);
+  //   setActiveMarker(marker);
+  //   setshowingInfoWindow(true);
+  //   setCurrProfile(profile)
+  //   // console.log("user is: ", user);
+  //   // setActiveProfile(user);
+  //   // calculateRouteBetweenUsers(user.address)
+  // };
+  const onClose = async (e) => {
+    if (showingInfoWindow) {
+      setshowingInfoWindow(false);
+      setActiveMarker(null);
+    }
+  };
 
   /** @type React.MutableRefObject<HTMLInputElement> */
-  const originRef = useRef()
+  const originRef = useRef();
   /** @type React.MutableRefObject<HTMLInputElement> */
-  const destiantionRef = useRef()
+  const destiantionRef = useRef();
 
   if (!isLoaded) {
-    return <SkeletonText />
+    return <SkeletonText />;
   }
 
   async function calculateRoute() {
-    if (originRef.current.value === '' || destiantionRef.current.value === '') {
-      return
+    if (originRef.current.value === "" || destiantionRef.current.value === "") {
+      return;
     }
     // eslint-disable-next-line no-undef
-    const directionsService = new google.maps.DirectionsService()
+    const directionsService = new google.maps.DirectionsService();
     const results = await directionsService.route({
       origin: originRef.current.value,
       destination: destiantionRef.current.value,
       // eslint-disable-next-line no-undef
       travelMode: google.maps.TravelMode.DRIVING,
-    })
-    setDirectionsResponse(results)
-    setDistance(results.routes[0].legs[0].distance.text)
-    setDuration(results.routes[0].legs[0].duration.text)
+    });
+    console.log("route is :", results)
+    setDirectionsResponse(results);
+    setDistance(results.routes[0].legs[0].distance.text);
+    setDuration(results.routes[0].legs[0].duration.text);
+  }
+  async function calculateRouteBetweenUsers(profileLocation) {
+  console.log("called")
+    if (originString  === "" || String(profileLocation) === "") {
```

Comment:
would check for nulls on profileLocation as well

Reply:
fixed